File: A/00/A07/A07.py
d = int(input())
n = int(input())
l = [None] * n
r = [None] * n

for i in range(n):
	l[i], r[i] = map(int, input().split())

data = [0] * (d + 2)
for i in range(n):
	data[l[i]] += 1
	data[r[i] + 1] -= 1

ans = [None] * (d + 2)
ans[0] = 0
for i in range(1, d + 1):
	ans[i] = ans[i - 1] + data[i]

for i in range(1, d + 1):
	print(ans[i])

File: A/00/A07/A07.c
#include <stdio.h>
#define N 100009

int main(void){
	int d, n, l[N], r[N], data[N], ans[N];

	scanf("%d%d", &d, &n);
	for(int i = 1; i <= n; i++)
		scanf("%d%d", &l[i], &r[i]);

	for(int i = 1; i <= n; i++){
		data[l[i]] += 1;
		data[r[i] + 1] -= 1;
	}

	ans[0] = 0;
	for(int i = 1; i <= d; i++)
		ans[i] = ans[i - 1] + data[i];

	for(int i = 1; i <= d; i++)
		printf("%d\n", ans[i]);
	return (0);
}
